dts/msm8974/msm8974-samsung-klte.dts

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+/include/ "skeleton.dtsi"
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+
+	// qcom,board-id = <0xCE08FF01 8>;
+	// ^^ no qcom,board-id in downstream DTB; downstream DTB has only this:
+	qcom,msm-id = <0xC208FF01 14 0x10000>;
+	// model = "SAMSUNG K PROJECT REV14";
+
+	klte {
+		model = "Samsung Galaxy S5 (SM-G900F)";
+		compatible = "samsung,klte", "qcom,msm8974", "lk2nd,device";
+		lk2nd,match-cmdline = "* samsung.hardware=SM-G900F *";
```

Comment:
Based on your log `lk2nd,match-bootloader = "G900F*";` should also work, I think that's a bit nicer for consistency with the msm8916 Samsung devices.

Reply:
It works